### ISSUE TITLE
Get error when tile generator in different tile shape

### ISSUE THREAD

**Issue by peter850421** [cgohlke/tifffile#38]
Hi,
Below is the tile generator sample code from the readme. The shape and tile shape are matches when the shape is times of tile, but when the shape is not divisible by tile, it will give an error. Need your help in dealing with this case. Thanks.

```
def tiles():
    data = numpy.arange(3*4*16*16, dtype='uint16').reshape((3*4, 16, 16))
    for i in range(data.shape[0]): yield data[i]
imwrite('temp.tif', tiles(), dtype='uint16', shape=(48, 64), tile=(16, 16))
```

-------Not divisible.------------

```
def tiles(data, tileshape):
    for y in range(0, 50, tileshape[0]):
        for x in range(0, 64, tileshape[1]):
            y2 = min(y + tileshape[0], 50)
            x2 = min(x + tileshape[1], 64)
            yield (data[:y2-y, :x2-x]).astype(np.uint8)
imwrite('temp.tif', tiles(data, (16,16)), dtype='uint16', shape=(50, 64), tile=(16, 16))
```

ValueError: invalid tile shape or dtype

**Comment by cgohlke** [comment]
All tiles in TIFF must be full size. For example:
```Python
import numpy
import tifffile

data = numpy.arange(50 * 64, dtype='uint16').reshape((50, 64))


def tiles(data, tileshape):
    for y in range(0, data.shape[0], tileshape[0]):
        for x in range(0, data.shape[1], tileshape[1]):
            tile = data[y : y + tileshape[0], x : x + tileshape[1]]
            if tile.shape != tileshape:
                tile = numpy.pad(
                    tile,
                    (
                        (0, tileshape[0] - tile.shape[0]),
                        (0, tileshape[1] - tile.shape[1])
                    )
                )
            yield tile


tifffile.imwrite(
    'issue38.tif',
    tiles(data, (16, 16)),
    dtype=data.dtype,
    shape=data.shape,
    tile=(16, 16)
)
```

**Comment by peter850421** [comment]
Thanks for your quick reply, and it's is interesting. This shape setting determines the output size, and it doesn’t matter if the tiles exceed this output size. 
Ask you further, whether the tiles function must traverse all areas in the output tiff or we can just build the output tiff only by put some tiles in certain areas and remaining blocks we can quickly set to zero or 255.

**Comment by cgohlke** [comment]
You must provide all tiles in correct order. If a tile is `None`, tifffile writes a tile of all zeros. There is no support for writing "sparse" TIFFs, e.g. as found in GeoTIFF.

**Comment by cgohlke** [comment]
This issue should be fixed in v2020.11.26.
